src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -180,10 +184,52 @@ MulticopterAttitudeControl::generate_attitude_setpoint(const Quatf &q, float dt,
 	// Align the desired tilt with the yaw setpoint
 	Quatf q_sp = q_sp_yaw * q_sp_rp;
 
-	q_sp.copyTo(attitude_setpoint.q_d);
-
 	// Transform to euler angles for logging only
+	Eulerf euler_des(q_sp);
```

Comment:
It would be prefered to do the calculations as quaternion directly or axis angle like in the stick generation.